src/core/meshcop/joiner.cpp

```diff
@@ -143,6 +143,9 @@ otError Joiner::Start(const char *     aPskd,
     otLogInfoMeshCoP("Joiner starting");
 
     VerifyOrExit(mState == OT_JOINER_STATE_IDLE, error = OT_ERROR_BUSY);
+    // Verify IPv6 stack is down and Thread stack is not fully enabled.
+    VerifyOrExit(Get<ThreadNetif>().IsUp() && Get<Mle::Mle>().GetRole() == Mle::kRoleDisabled,
```

Comment:
We may also consider start IPv6 stack when it's not enabled for convenience. Thoughts?

Reply:
Sounds like a good idea to me (to start the stack if not already up). Make it easier to use.

Reply:
updated!